packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -120,32 +126,16 @@ export default createESLintRule<Options, MessageIds>({
       return {};
     }
 
-    // gather helper dependencies for @nx/js executors
-    const helperDependencies = getHelperDependenciesFromProjectGraph(
-      workspaceRoot,
-      sourceProject.name,
-      projectGraph
-    );
-
     const rootPackageJson = getPackageJson(join(workspaceRoot, 'package.json'));
 
-    // find all dependencies for the project
-    const npmDeps = findProjectsNpmDependencies(
+    const npmDependencies = findNpmDependencies(
+      workspaceRoot,
       sourceProject,
       projectGraph,
-      buildTarget,
```

Comment:
Why not pass in `buildTarget` so we can get a file set for filtered files (instead of explicit ignoreFilePatterns?